scripts/validate_version.py

```diff
@@ -1,13 +1,22 @@
 import re
 import sys
+from packaging import version
 
 # semantic version regex
 regex = "^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
-result = re.search(regex, sys.argv[1])
+oldversion = sys.argv[1]
+newversion = sys.argv[2]
+regex_match = re.search(regex, sys.argv[1])
+# compare the oldversion and newversion
+version_compare = version.parse(oldversion) < version.parse(newversion)
```

Comment:
I think this comparison is not required when we are increasing the version automatic.